Source/Dafny/Resolver.cs

```diff
@@ -11403,6 +11403,7 @@ private class RBranchExpr : RBranch {
         for (int id = 0; id < mti.BranchIDCount.Length; id++) {
           if (mti.BranchIDCount[id] <= 0) {
             reporter.Warning(MessageSource.Resolver, mti.BranchTok[id], "this branch is redundant ");
+            ResolveExpression(e.Cases.ElementAt(id).Body, new ResolveOpts(codeContext, false)); // TODO: what is the correct value for IsTwoState
```

Comment:
Change the surrounding method (`CompileNestedMatchExpr`) so that instead of taking an `ICodeContext codeContext`, it takes a `ResolveOpts opts`. Then, just pass in that `opts` here.

Reply:
Fixed